Code/Tools/AssetProcessor/native/ui/BuilderListModel.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <native/ui/BuilderListModel.h>
+#include <utilities/AssetUtilEBusHelper.h>
+
+int BuilderListModel::rowCount(const QModelIndex& /*parent*/) const
+{
+    AssetProcessor::BuilderInfoList builders;
+    AssetProcessor::AssetBuilderInfoBus::Broadcast(&AssetProcessor::AssetBuilderInfoBus::Events::GetAllBuildersInfo, builders);
+
+    return aznumeric_caster(builders.size());
+}
+
+QVariant BuilderListModel::data(const QModelIndex& index, int role) const
+{
+    AssetProcessor::BuilderInfoList builders;
+    AssetProcessor::AssetBuilderInfoBus::Broadcast(&AssetProcessor::AssetBuilderInfoBus::Events::GetAllBuildersInfo, builders);
+
+    AZ_Assert(index.row() >= 0, "Index must be >= 0");
+    AZ_Assert(index.row() < builders.size(), "Index out of bounds");
+
+    const auto& assetBuilderDesc = builders[index.row()];
+
+    if (role == Qt::DisplayRole)
+    {
+        return QString(assetBuilderDesc.m_name.c_str());
+    }
+
+    return {};
+}
+
+void BuilderListModel::Reset()
+{
+    beginResetModel();
```

Comment:
usually you call beginResetModel before adjusting any internal state (so that views know not to query the model now) and endResetModel once you're done. Calling these functions without anything in between worries me. When is this supposed to be used? I know this is more of an adapter than a standalone model, but is it possible to do any necessary state management between these two?

Reply:
So I couldn't find a one-off signal to say "hey the model updated, please regenerate the view", this seemed to be the closest I could find.  I had to do this because this model technically doesn't own any data, its just interfacing with the AP where the data is actually owned.  Maybe there are better ways to do this though, I'm not too familiar with Qt.

Reply:
ok, sounds good. Yeah, that's what reset is for, but ideally you let the view know that it's going to happen before you start (so it doesn't query anything while you work) and then let it know when you're finished, but I don't think there's much you can do about it in this case.